code/modules/projectiles/projectile/bullets/smg.dm

```diff
@@ -50,3 +50,34 @@
 	name = "4.6x30mm rubber bullet"
 	damage = 5
 	stamina = 22
+
+/obj/item/projectile/bullet/c46x30mm/venom
+	name = "4.6x30mm venom bullet" 
+	damage = 6
+
+/obj/item/projectile/bullet/c46x30mm/venom/on_hit(atom/target, blocked)
+	if((blocked != 100) && iscarbon(target))
+		var/mob/living/carbon/victim = target // Both injects toxin, and applies 6 tox damage on hit.
+		victim.reagents.add_reagent(/datum/reagent/toxin, 4)
+		victim.adjustToxLoss(6)
+
+	return ..()
+/obj/item/projectile/bullet/c46x30mm/kraken
+	name = "4.6x30mm kraken bullet"
+	damage = 22
+	armour_penetration = -60
+	wound_bonus = -30 // we arent dismembering people here
+	bare_wound_bonus = 3
+
+/obj/item/projectile/bullet/c46x30mm/airburst_pellet
+	name = "4.6x30mm airburst pellet"
+	damage = 10
+/obj/item/projectile/bullet/c46x30mm/airburst
+	name = "4.6x30mm airburst bullet"
+	damage = 2 // its just a casing
+	range = 6
+
+/obj/item/projectile/bullet/c46x30mm/airburst/on_range(obj/item/projectile/bullet/c46x30mm/airburst/A, mob/living/carbon/human/D)
+	var/obj/item/ammo_casing/c46x30mm/airburst_pellet/P = new(get_turf(src))
+	P.fire_casing(D, A)
+	..()
```

Comment:
have you tested this? does it work?

Reply:
but i was given those variables from elsewhere in the code, on_range isn't passed those parameters